### ISSUE TITLE
executable won't be generated after clean

### ISSUE THREAD

**Issue by sapei** [gogradle/gogradle#217]
In gogradle v0.9, the executable file is not going to be generated again if clean task has been performed before and there is no changes in the source codes.

To reproduce:
- Execute the build task
- Afterwards perform the clean task
- Execute again the build task. 
In this case the task ':buildLinuxAmd64' is considered to be up to date.

How to enforce the plugin to generate the executable if it does not exist, even though there are no changes in the codes (without using the flag --rerun-tasks)? Would it be possible to add extra check in the task buildLinuxAmd64 using the method [outputs.upToDateWhen](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskOutputs.html#upToDateWhen-groovy.lang.Closure-)?

**Comment by baron1405** [comment]
I experienced the same problem and had to delete the .gradle directory at the root of the project to have the build regenerate the executable. My configuration is Linux, Gradle 4.7, GoGradle 0.9.

**Comment by baron1405** [comment]
It appears to be working now. Perhaps it is cached state in the project's .gradle directory.

**Comment by baron1405** [comment]
It is actually intermittent. Sometimes it won't work at all and then it works fine for a while. Looking at the debug output, the platform specific build task says it is up-to-date even when the output file is missing.

**Comment by blindpirate** [comment]
Thanks @baron1405 ! I'll investigate it today.

**Comment by baron1405** [comment]
Hi @blindpirate, find anything?

**Comment by blindpirate** [comment]
Thanks for reporting! I'm wondering how you configure your `GoBuild` task. If it's configured by:

```
build {
     go 'build -o ./outputFile github.com/my/package'
}
```

Then gogradle doesn't register `outputFile` as output. In this case, the output location should be specified explicitly:

```
build {
     go 'build -o ./outputFile github.com/my/package'
     outputLocation = './outputFile'
}
```

**Comment by baron1405** [comment]
First, to avoid conflicts with existing "build" tasks, I set:
```
systemProp.gogradle.alias=true
```
in my ~/.gradle/gradle.properties. Then in my build.gradle.kts, I configure the "goBuild" task with just the outputLocation. I have the existing "build" task depend on the "goBuild" task:
```
"goBuild" {
    val buildTask = this as GoBuild
    buildTask.outputLocation = "build/apron-\${GOOS}"
}

"build" { dependsOn("goBuild") }
```
Note that the above is in Kotlin.

I do not explicitly invoke "go 'build ...'". Doesn't the "goBuild" task do that? Am I supposed to explicitly inoke "go" to build my app?

**Comment by baron1405** [comment]
Thanks for the quick response @blindpirate!

**Comment by blindpirate** [comment]
Closed by `0.9.1`.
